abel/tests/test_tools_vmi.py

```diff
@@ -0,0 +1,146 @@
+        assert_allclose(speeds[2:R], ref(r[2:R]), rtol=rtol,
+                        err_msg='-> {}, {}'.format(name, kwargs))
+
+    # for Jacobian=True
+    def spher(r):
+        return 4 * r**2  # (in fact, must be 4 * np.pi * r**2)
+
+    check('ones', spher, 0.001, ones)
+    check('ones', spher, 0.001, ones, dr=0.5)
+    check('ones', spher, 0.0001, ones, dt=0.01)
+    check('cos2', spher, 0.01, cos2 * 3)
+    check('sin2', spher, 0.01, sin2 * 3/2)
+
+    # for Jacobian=False
+    def polar(r):
+        return np.full_like(r, 2 * np.pi)  # (in fact, should be 2 * np.pi * r)
+
+    check('ones', polar, 0.02, ones, Jacobian=False)
+    check('ones', polar, 0.02, ones, Jacobian=False, dr=0.5)
+    check('ones', polar, 0.01, ones, Jacobian=False, dt=0.01)
+    check('cos2', polar, 0.03, cos2 * 2, Jacobian=False)
+    check('sin2', polar, 0.001, sin2 * 2, Jacobian=False)
+
+
+def test_average_radial_intensity():
+    """
+    Basic tests of angular averaging.
+    """
+    R = 50  # image radius
+    n, ones, cos2, sin2 = make_images(R)
+
+    def check(name, ref, atol, IM, **kwargs):
+        r, intensity = average_radial_intensity(IM, **kwargs)
+        assert_allclose(intensity[2:R], ref, atol=atol,
+                        err_msg='-> {}, {}'.format(name, kwargs))
+
+    check('ones', 1, 0.02, ones)
+    check('ones', 1, 0.02, ones, dr=0.5)
+    check('ones', 1, 0.01, ones, dt=0.01)
+    check('cos2', 1/2, 0.02, cos2)
+    check('sin2', 1/2, 0.001, sin2)
+
+
+def test_anisotropy_parameter():
+    """
+    Basic tests of anisotropy fitting.
+    """
+    n = 100
+    theta = np.linspace(-np.pi, np.pi, n, endpoint=False)
+
+    ones = np.ones_like(theta)
+    cos2 = np.cos(theta)**2
+    sin2 = np.sin(theta)**2
+
+    def check(name, ref, theta, intensity):
+        beta, amplitude = anisotropy_parameter(theta, intensity)
+        assert_allclose((beta[0], amplitude[0]), ref, atol=1e-8,
+                        err_msg='-> ' + name)
+
+    check('ones', (0, 1), theta, ones)
+    check('cos2', (2, 1/3), theta, cos2)
+    check('sin2', (-1, 2/3), theta, sin2)
+    check('cos2sin2', (0, 1/8), theta, cos2 * sin2)
+
+
+def test_radial_integration():
+    """
+    Basic test of radial integration.
+    """
+    # test image (not projected)
+    IM = SampleImage(name='dribinski').image
+
+    Beta, Amplitude, Rmidpt, _, _ = \
+        radial_integration(IM, radial_ranges=([(65, 75), (80, 90), (95, 105)]))
+
+    assert_equal(Rmidpt, [70, 85, 100], err_msg='Rmidpt')
+    assert_allclose([b[0] for b in Beta], [0, 1.58, -0.85], atol=0.01,
+                    err_msg='Beta')
+    assert_allclose([a[0] for a in Amplitude], [880, 600, 560], rtol=0.01,
+                    err_msg='Amplitude')
+
+
+def test_toPES():
+    """
+    Check that toPES at least does not crash.
+    TODO: do some meaningful tests.
+    """
+    # test image (not projected)
+    IM = SampleImage(name='dribinski').image
```

Comment:
`IM = SampleImage(name='Ominus').image`, the pseudo O<sup>-</sup> photoelectron spectrum would be more realistic for this test, and the following parameters give something close to reality, with the dominant peak positioned around 11784.68 cm<sup>-1</sup>, the electron affinity of oxygen:
```python
    # test image (not projected)
    IM = SampleImage(name='Ominus').image

    # (parameters from example_hansenlaw.py, although for a different image)
    eBE, PES = toPES(*angular_integration(IM),
                     energy_cal_factor=1.2e-5,
                     photon_energy=1.0e7/808.6, Vrep=-100,
                     zoom=IM.shape[-1]/2048)

    assert_allclose(eBE[PES.argmax()], 11784.68, 0.1)

```
![Screenshot-20210428084442-588x417](https://user-images.githubusercontent.com/10932229/116321748-00b00700-a7fe-11eb-8782-bb97b3ac9114.png)

`n=1001` would be better for realism, but not essential for this test.